### PR TITLE
Add `See also: `, remove description of other functions in base.array.jl

### PR DESCRIPTION
**Pull request by ChristianKurz**
#23788

### CODE REVIEW

File: base/array.jl [hunk 1]
```diff
@@ -1080,6 +1080,8 @@ end
 Remove an item in `collection` and return it. If `collection` is an
 ordered container, the last item is returned.
 
+See also: shift!, splice!
```
Comment (KristofferC) [inline base/array.jl]:
```
[`shift!`](@ref)
```

etc

File: base/array.jl [hunk 2]
```diff
@@ -1080,6 +1080,8 @@ end
 Remove an item in `collection` and return it. If `collection` is an
 ordered container, the last item is returned.
 
+See also: [`shift!`](@ref), [`splice!`](@ref)
```
Comment (fredrikekre) [inline base/array.jl]: What about `push!`

Comment (shorty66) [reply base/array.jl]: Push! adds elements, whereas the other functions remove elements. I think, `See also:`-lists should be rather minimal: They reduce the need to search the documentation but it's no replacement for a proper search.

Comment (StefanKarpinski) [reply base/array.jl]: A thesaurus lists both similar words and antonyms for a reason – knowing about `push!` is useful if you're looking at `pop!`. From `push!` you could potentially find all the other element addition functions.

File: base/array.jl [hunk 3]
```diff
@@ -1147,6 +1149,8 @@ end
 
 Remove the first `item` from `collection`.
 
+See also: [`pop!`](@ref), [`splice!`](@ref)
```
Comment (garrison) [inline base/array.jl]: What about `unshift!`?

File: base/array.jl [hunk 4]
```diff
@@ -982,6 +991,10 @@ julia> prepend!([3],[1,2])
  2
  3
 ```
+
+Use [`unshift!`](@ref) to add individual items to `collection` which are not already
+themselves in another collection. The result is of the preceding example is equivalent to
```
Comment (garrison) [inline base/array.jl]: I find the phrase "which are not already themselves in another collection" confusing.  I think I know what you are getting at, but I would remove it.

File: base/array.jl [hunk 5]
```diff
@@ -982,6 +991,10 @@ julia> prepend!([3],[1,2])
...
+`unshift!([1, 2, 3], 4, 5, 6)`.
```
Comment (garrison) [inline base/array.jl]: The previous example (`prepend!([3],[1,2])` I believe) actually has a different result.  And there is a duplicate "is" in the above sentence.

File: base/array.jl [hunk 6]
```diff
@@ -895,9 +898,8 @@ julia> push!([1, 2, 3], 4, 5, 6)
  6
 ```
 
-Use [`append!`](@ref) to add all the elements of another collection to
```
Comment (nalimilan) [inline base/array.jl]: Please don't change unrelated lines. This applies to other places below.

File: base/array.jl [hunk 7]
```diff
@@ -974,14 +978,29 @@ end
...
+6-element Array{Int64,1}:
+ 4
+ 5
+ 6
+ 1
+ 2
+ 3
 ```
+
+Use [`unshift!`](@ref) to add individual items to `collection`. The result of
```
Comment (nalimilan) [inline base/array.jl]: I'd rather put this in the "See also" paragraph, rather than after the examples, where people might not find it (examples can be long). No need to refer to a particular example, just say "`prepend!(x, y)` is equivalent to...".

Comment (shorty66) [reply base/array.jl]: I removed these sentences in favor of a "See also:" entry. We opted against further descriptions in "See also" earlier.

File: base/array.jl [hunk 8]
```diff
@@ -916,9 +916,11 @@ function push!(a::Array{Any,1}, @nospecialize item)
 end
 
 """
-    append!(collection, collection2) -> collection.
+    append!(a::Vector, items) -> collection.
```
Comment (fredrikekre) [inline base/array.jl]: Why this change?

File: base/array.jl [hunk 9]
```diff
@@ -974,13 +972,24 @@ end
...
+
 # Examples
 ```jldoctest
 julia> prepend!([3],[1,2])
 3-element Array{Int64,1}:
  1
  2
  3
+
+julia> prepend!([1, 2, 3], [4, 5, 6])
```
Comment (fredrikekre) [inline base/array.jl]: What does this extra example show that the one just above doesn't?

File: base/array.jl [hunk 10]
```diff
@@ -1212,6 +1229,8 @@ end
 Remove the item at the given `i` and return the modified `a`. Subsequent items
 are shifted to fill the resulting gap.
 
+See also: [`splice!`](@ref)
```
Comment (nalimilan) [inline base/array.jl]: Maybe also `pop!` and `unshift!`?

File: base/array.jl [hunk 11]
```diff
@@ -1316,6 +1335,8 @@ Subsequent items are shifted left to fill the resulting gap.
 If specified, replacement values from an ordered
 collection will be spliced in place of the removed item.
 
+See also: [`pop!`](@ref), [`shift!`](@ref), [`insert!`](@ref)
```
Comment (nalimilan) [inline base/array.jl]: Also `deleteat!` (i.e. links should generally go both directions, right?).

File: base/array.jl (hunk 8 above)
Comment (shorty66) [reply base/array.jl]: `a` has to be a vector for `append!` to work. Of course, `items` could stay `collection`, but this wording is consistent with the other docstrings. Perhaps `collection` would be clearer, though, as items may be interpreted as several seperate items and `append!` expects a collection of items...

```julia
julia> append!([1 2 3], 1)
ERROR: MethodError: no method matching append!(::Array{Int64,2}, ::Int64)

You might have used a 2d row vector where a 1d column vector was required.
Note the difference between 1d column vector [1,2,3] and 2d row vector [1 2 3].
You can convert to a column vector with the vec() function.
Closest candidates are:
  append!(::Array{T,1} where T, ::Any) at array.jl:637

julia> append!([1,2,3], 1)
4-element Array{Int64,1}:
 1
 2
 3
 1
```

File: base/array.jl (hunk 9 above)
Comment (shorty66) [reply base/array.jl]: this is a remnant of earlier edits, will remove.

File: base/array.jl (hunk 11 above)
Comment (shorty66) [reply base/array.jl]: Yeah, right. I think this would be easier if `See also:` lists would be generated automatically from some kind of relation-table.

File: base/array.jl (hunk 8 above)
Comment (nalimilan) [reply base/array.jl]: This change should go to a separate commit or PR, and it should use `AbstractVector`. I also wonder whether this isn't on purpose, since non-`AbstractVector` collections defined in packages could define methods and it would be silly to repeat the information in another docstring.
